flang-rt/lib/runtime/CMakeLists.txt

```diff
@@ -183,6 +183,10 @@ endif ()
 
 
 if (NOT WIN32)
+  add_definitions(-U_GLIBCXX_ASSERTIONS -D_GLIBCXX_NO_ASSERTIONS)
+  add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-fno-exceptions>)
+  add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-O2>)
```

Comment:
1. `-U_GLIBCXX_ASSERTIONS` and `fno-exceptions` are already added in AddFlangRT.cmake. `_GLIBCXX_NO_ASSERTIONS` should not be necessary, but could be an alternative to  `-U_GLIBCXX_ASSERTIONS`
2. We prefer [modern CMake](https://hsf-training.github.io/hsf-training-cmake-webpage/04-targets/index.html) style
3. Adding `-O2` overrides the choice of `CMAKE_BUILD_TYPE`. What optimization level to use should be left to the user.

Reply:
@Meinersbur

Re 2: A suggested alternative syntax would be helpful in understanding what you are suggesting.
Re 3: I don't like forcing `-O2` either, but I don't see an alternative to making the DCE happen that's necessary to get rid of the offending undefined symbol.

Reply:
[`target_compile_options`](https://cmake.org/cmake/help/latest/command/target_compile_options.html) and [`target_compile_definitions`](https://cmake.org/cmake/help/latest/command/target_compile_definitions.html). See [AddFlangRT.cmake](https://github.com/llvm/llvm-project/blob/main/flang-rt/cmake/modules/AddFlangRT.cmake). `_GLIBCXX_ASSERTIONS` and `-fno-exceptions` are already handled there. If there are any issus with those, it should be fixed in `AddFlangRT.cmake`, not here.